book/advanced/prove-scripts.md

```diff
@@ -0,0 +1,174 @@
+
+### When to Use Aggregation
+
+Aggregation is useful when you need to:
+- Reduce on-chain verification costs by combining multiple range proofs into one proof
+- Prove larger block ranges by aggregating individual range proofs from different time periods
+- Combine proofs from different proving sessions into a single proof for batch submission
+
+> **Note:** All range proofs must be generated in compressed mode for aggregation. The prove scripts handle this automatically.
+
+## Setup
+
+### Environment Configuration
+
+Create a `.env` file in the project root directory:
+
+```bash
+# RPC Endpoints
+L1_RPC=<YOUR_L1_RPC_ENDPOINT>
+L1_BEACON_RPC=<YOUR_L1_BEACON_RPC_ENDPOINT>
+L2_RPC=<YOUR_L2_RPC_ENDPOINT>
+L2_NODE_RPC=<YOUR_L2_NODE_RPC_ENDPOINT>
+
+# Network Prover Configuration
+NETWORK_PRIVATE_KEY=<YOUR_NETWORK_PRIVATE_KEY>
+
+# Proof Strategy Configuration
+RANGE_PROOF_STRATEGY=reserved    # Options: reserved, hosted, auction
+AGG_PROOF_STRATEGY=reserved      # Options: reserved, hosted, auction
+AGG_PROOF_MODE=plonk             # Options: plonk, groth16
+```
+
+### Environment Variables
+
+#### Required
+
+| Variable | Description |
+|----------|-------------|
+| `L1_RPC` | L1 Archive Node endpoint |
+| `L1_BEACON_RPC` | L1 Consensus (Beacon) Node endpoint |
+| `L2_RPC` | L2 Execution Node (`op-geth`) endpoint |
+| `L2_NODE_RPC` | L2 Rollup Node (`op-node`) endpoint |
+| `NETWORK_PRIVATE_KEY` | Private key for the Succinct Prover Network. See the [Succinct Prover Network Quickstart](https://docs.succinct.xyz/docs/sp1/prover-network/quickstart) for setup instructions. |
+
+#### Optional
+
+| Variable | Description | Default |
+|----------|-------------|---------|
+| `RANGE_PROOF_STRATEGY` | Proof fulfillment strategy for range proofs | `reserved` |
+| `AGG_PROOF_STRATEGY` | Proof fulfillment strategy for aggregation proofs | `reserved` |
+| `AGG_PROOF_MODE` | Proof mode for aggregation proofs (`plonk` or `groth16`) | `plonk` |
+
+**Proof Strategies:**
+- `reserved`: Uses reserved SP1 network capacity
+- `hosted`: Uses hosted proof generation service
+- `auction`: Uses auction-based proof fulfillment
+
+**Proof Modes:**
+- `plonk`: PLONK proof system (default)
+- `groth16`: Groth16 proof system
+
+### Getting Started with the Prover Network
+
+1. Follow the [Succinct Prover Network Quickstart](https://docs.succinct.xyz/docs/sp1/prover-network/quickstart) to set up your account and obtain a private key.
+
+2. Set the `NETWORK_PRIVATE_KEY` environment variable:
+   ```.env
+   NETWORK_PRIVATE_KEY=0x...
+   ```
+
+3. Run the prove scripts. The binaries will automatically use the network prover with your configured key.
+
+## Generating Range Proofs
+
+The `multi.rs` binary generates compressed range proofs for a specified block range. These proofs verify the state transition function for the L2 blocks in the range.
+
+### Usage
+
+```bash
+cargo run --bin multi --release -- \
+    --start <START_BLOCK> \
+    --end <END_BLOCK> \
```

Comment:
Based on what we found yesterday, `--batch-size` should always match `end - start` for the command to work correctly. When both end and start are provided, we should default the batch size to that value. We can handle that in a separate PR.